chromium_src/chrome/browser/ui/startup/startup_browser_creator_impl.h

```diff
@@ -0,0 +1,17 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_STARTUP_STARTUP_BROWSER_CREATOR_IMPL_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_STARTUP_STARTUP_BROWSER_CREATOR_IMPL_H_
+
+// Make BraveStartupBrowserCreatorImpl a friend class, to give it access to
+// StartupBrowserCreatorImpl's private attributes.
+#define browser_creator_ \
+  browser_creator_;      \
```

Comment:
this seems a bit fragile to me because we'll have no way to know it stops working if `browser_creator_` goes away or changes the name. We definitely need a test for this.

Reply:
If it goes away, compilation will definitely fail

Reply:
oh, because it's calling private methods? Right, ok

Reply:
still seems like we should have a test though

Reply:
I added a browser test in 43e9e5d03eb8064c5d72295cc06a9c85e0168f0e. As suggested by @bridiver, it's a simple adaptation of `SecondLaunchWithIncognitoUrl`. It launches a new window with the `--tor` switch and makes sure that this results in an additional window.